controls/mousestates/cMouseState.h

```diff
@@ -0,0 +1,32 @@
+#ifndef D2TM_CMOUSESTATE_H
+#define D2TM_CMOUSESTATE_H
+
+#include "observers/cInputObserver.h"
+#include "controls/cMouse.h"
+//#include "controls/cGameControlsContext.h"
+
+// forward declaration
+class cGameControlsContext;
+class cPlayer;
+
+class cMouseState : public cInputObserver {
+
+public:
+    cMouseState(cPlayer * player, cGameControlsContext *context, cMouse * mouse);
+    ~cMouseState();
```

Comment:
This destructor can be left out completely. The compiler will generate the correct default.

Reply:
👍